proto/identity/associations/association.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package xmtp.identity.associations;
+
+import "identity/associations/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/identity/associations";
+option java_package = "org.xmtp.proto.identity.associations";
+
+// The first entry of any XID log, which is considered valid iff the XID can be
+// deterministically derived from the address and nonce.
+// The recovery address defaults to the initial_associated_address unless
+// there is a subsequent ChangeRecoveryAddress in the log.
+message CreateXid {
+  string xid = 1;
+  uint32 nonce = 2;
+  string initial_associated_address = 3;
+}
+
+message AddAssociation {
+  enum AssociatedRole {
+    ASSOCIATED_ROLE_UNSPECIFIED = 0;
+    ASSOCIATED_ROLE_INSTALLATION = 1;
+    ASSOCIATED_ROLE_ADDRESS = 2;
+    ASSOCIATED_ROLE_LEGACY_KEY = 3;
+  }
+
+  string xid = 1;
```

Comment:
The reason I had this field in the API request instead of here is because I'm imagining these whole entries are going to get serialized into the credential and stored in the DB. Less duplication that way, although it's obviously a small optimization in the scheme of things.

Reply:
Are you referring to the server DB or client DB? And with the new design, credentials will only contain an XID, sequence number, and public installation key right?

My main thinking with placing the XID here is that for constructing the signature text, all of the information should be in one place

Reply:
I was thinking about the server DB records, since the xid is going to be stored outside of this payload somewhere so it can be indexed. 

My comment about the credential is out of date. I do think all we need in the credential are the things you listed above.